code/game/turfs/simulated/floor/asteroid_floors.dm

```diff
@@ -361,85 +198,6 @@ GLOBAL_LIST_INIT(megafauna_spawn_list, list(/mob/living/simple_animal/hostile/me
 		O.forceMove(pick_n_take(valid_turfs))
 	qdel(src)
 
-/turf/simulated/floor/plating/asteroid/airless/cave/proc/SpawnFloor(turf/T, monsterprob = 30)
-	if(istype(T.loc, /area/ruin))
-		return
-
-	for(var/S in RANGE_TURFS(1, src))
-		var/turf/NT = S
-		if(!NT || isspaceturf(NT) || istype(NT.loc, /area/lavaland/surface/outdoors/explored))
-			sanity = 0
-			break
-	if(!sanity)
-		return
-	SpawnFlora(T)
-
-	SpawnMonster(T, monsterprob)
-	T.ChangeTurf(turf_type, FALSE, FALSE, TRUE)
-
-/turf/simulated/floor/plating/asteroid/airless/cave/proc/SpawnMonster(turf/T, monsterprob = 30)
-	if(istype(T.loc, /area/ruin))
-		return
-
-	if(prob(monsterprob))
-		if(!istype(loc, /area/lavaland/surface/outdoors/unexplored))
-			return
-		var/randumb = pickweight(mob_spawn_list)
-		while(randumb == SPAWN_MEGAFAUNA)
-			if(istype(loc, /area/lavaland/surface/outdoors/unexplored/danger)) //this is danger. it's boss time.
-				var/maybe_boss = pickweight(megafauna_spawn_list)
-				if(megafauna_spawn_list[maybe_boss])
-					randumb = maybe_boss
-			else //this is not danger, don't spawn a boss, spawn something else
-				randumb = pickweight(mob_spawn_list)
-		var/scanrange = 12
-		var/megafaunarange = 7
-		switch(SSmapping.cave_theme)
-			if(DEADLY_DEEPROCK)
```

Comment:
You'll probably want to see about adding this function back in for deadly deeprock, seems missing from the rework so the big rooms won't spawn as many mobs

Reply:
Done, I think, PTAL